File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.c
int isPrefixOfWord(char* s, char* k) {
    int count = 1 ; 
    for(int i = 0 ;s[i]!=0 ; ){
        if(s[i]==' '||i==0){
            if(i)
                count ++ ; 
                if(i)
                i++ ;
                int x = 0 ; 
            while(k[x]!=0&&s[x+i]==k[x]){
                x++ ;
            }
            if(k[x]==0){
                return count ;
        }
        }else i++;
        if(i==0){
            i++ ;
        }
        
       
    }
    return -1 ;
}

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.java
class Solution {
    public int isPrefixOfWord(String sentence, String k) {
        String  [] s= sentence.split(" "); 
        for(int i = 0 ; i<s.length ; i++){
            if(s[i].startsWith(k)){
                return i+1 ;
            }
        } 
        return -1 ;
    }
}